### ISSUE TITLE
Incorrect Implementation of Dice Metric in Segmentation_Trainer in trainer_ddp.py

### ISSUE THREAD

**Issue by MasalaKimchi** [OSUPCVLab/SegFormer3D#8]
Thank you for providing the code open-source! Great job on presenting your work at CVPR 2024! 

In **trainer_ddp.py,** a class **Segmentation_Trainer** is used to train for segmentation.
Within class of **Segmentation_Trainer**, function **_val_step** is called for dice coefficient.
**You seem to be passing the image instead of the ground truth** 

Snippet of code from **_val_step**, passes variable **data** and **labels** to the function **self._calc_dice_metric**.
It seems like current code is using the **original image** and **labels** to compute dice metric, which does not make sense to me.

I think it should be fixed as below. Please let me know if my understanding is wrong.
```
mean_dice = self._calc_dice_metric(predicted, labels, use_ema) 
```

Source code for reference below
```
 with torch.no_grad():
      for index, (raw_data) in enumerate(self.val_dataloader):
          # get data ex: (data, target)
          data, labels = (
              raw_data["image"],
              raw_data["label"],
          )
          # forward pass
          if use_ema:
              predicted = self.ema_model.forward(data)
          else:
              predicted = self.model.forward(data)
          # calculate loss
          loss = self.criterion(predicted, labels)
          # calculate metrics
          if self.calculate_metrics:
              mean_dice = self._calc_dice_metric(data, labels, use_ema)    # This seems to be wrong
              # keep track of number of total correct
              total_dice += mean_dice

          # update loss for the current batch
          epoch_avg_loss += loss.item()
```

Thanks in advance,

**Comment by MasalaKimchi** [comment]
It was my misunderstanding on how sliding_window_inference worked, apologies for the confusion!
